File: low-level-design/design-patterns/golang/behavioural/chain-of-responsibility-pattern/cor/quota_check_handler.go
package cor

import (
	"errors"
	"fmt"
)

func NewQuotaCheckHandler(next Handler) *QuotaCheckHandler {
	handler := &QuotaCheckHandler{}
	handler.SetNext(next)
	return handler
}

type QuotaCheckHandler struct {
	BaseHandler
}

func (h *QuotaCheckHandler) Handle(request *DeploymentRequest) error {
	if request.CPU > 1000 || request.Memory > 2048 {
		fmt.Printf("Quota exceeded for service %s: CPU %d, Memory %d\n", request.ServiceName, request.CPU, request.Memory)
		return errors.New("quota exceeded")
	}
	fmt.Print("Quota check passed\n")
	return h.CallNext(request)

}

File: low-level-design/design-patterns/python/behavioural/chain-of-responsibility-pattern/quota_check_handler.py
from handler import BaseHandler,Handler,DeploymentRequest

class QuotaCheckHandler(BaseHandler):
    def __init__(self,handler:Handler):
        self.set_next(handler)
    
    def handle(self,request:DeploymentRequest):
        if request.cpu>10 or request.memory>20:
            print("QuotaCheckHandler: Request exceeds quota. Denying deployment.")
            return False
        print("QuotaCheckHandler: Quota check passed. Proceeding to next handler.")
        return self.call_next(request)
